src/builder.ts

```diff
@@ -78,16 +77,26 @@ export default class Builder {
 		Bluebird.resolve(this.docker.buildImage(inputStream, buildOpts))
 		.then((res: NodeJS.ReadWriteStream) => {
 
-			const outputStream = res
+			let errored = false;
+			// Create a stream to write parsed output from the docker daemon
+			const outputStream = new Stream.PassThrough();
 			// parse the json objects
-			.pipe(JSONStream.parse())
-			// Don't use fat-arrow syntax here, to capture 'this' from es
-			.pipe(es.through(function(data: any): void {
+			res.pipe(JSONStream.parse())
+			.on('data', (data: any) => {
+				// data object examples:
+				// { stream: 'Step 1/6 : FROM resin/raspberry-pi-node:0.10.22\n' }
+				// { status: 'Pulling from resin/raspberry-pi-node', id: '0.10.22' }
+				// { status: 'Pulling fs layer', progressDetail: {}, id: '6b5c2e9bbf98' }
+				// { status: 'Verifying Checksum', progressDetail: {}, id: 'e4d99e807699' }
+				// { status: 'Downloading', id: '6b5c2e9bbf98',
+  				//   progressDetail: { current: 1792780, total: 2271460 },
+  				//   progress: '[=======================================>           ]  1.793MB/2.271MB' }
 				if (data == null) {
 					return
 				}
 				if (data.error) {
-					dup.destroy(new Error(data.error))
+					// data.error example: 'Error: image busybox:latest not found'
+					outputStream.emit('error', new Error(data.error));
```

Comment:
The previous code here was `dup.destroy(new Error(data.error))`. With a separate outpuStream, I found it more intuitive to "forward" the error from the JSONStream to the outputStream. The documentation of duplexify reads:
> _If the readable or writable streams emits an error or close it will destroy both streams and bubble up the event. You can also explicitly destroy the streams by calling `dup.destroy()`._

So `outputStream.emit('error')` will trigger the `dup.on('error')` event, which was tested and is relied on a few lines below.